src/executor/Executor03BytecodeBuilder.ts

```diff
@@ -147,6 +147,7 @@ export class Executor03BytecodeBuilder extends ExecutorBytecodeBuilder<
     if (
       flags.dexes[index] % 4 !== 1 && // not sendEth
       !isETHAddress(swap.srcToken) &&
+      !curExchangeParam.skipApproval &&
```

Comment:
same here for the next approval before deposit

Reply:
fixed